frontend_tests/casper_tests/08-edit.js

```diff
@@ -84,6 +84,46 @@ casper.waitWhileVisible("textarea.message_edit_content", function () {
     casper.test.assertSelectorHasText(".last_message .message_content", "test edited pm");
 });
 
+// test editing last own message
+// 37 is left arrow key code
+casper.then(function () {
+    casper.test.assertNotVisible('form.message_edit_form', 'Message edit box not visible');
+
+    common.keypress(37);
+    casper.waitUntilVisible("#message_edit_content", function () {
+        var fieldVal = common.get_form_field_value('#message_edit_content');
+        casper.test.assertEquals(fieldVal, "test edited pm", "Opened editing last own message");
+        casper.click('.message_edit_cancel');
+    });
+});
+
+casper.waitWhileVisible('.message_edit', function () {
+    casper.click('body');
+    casper.page.sendEvent('keypress', "c");
+    casper.waitUntilVisible('#compose', function () {
+        casper.evaluate(function () {
+            $('#new_message_content').expectOne().focus();
+            $('#new_message_content').trigger($.Event('keydown', { which: 37 }));
+        });
+        casper.waitUntilVisible(".message_edit_form", function () {
+            casper.echo("Opened editing last own message");
+            casper.click('.message_edit_cancel');
+        });
```

Comment:
`casper.waitUntilVisible`  shoudln't be nested; put this in a separate `casper.then()` block to avoid races.